qunfold/transformers.py

```diff
@@ -133,22 +133,19 @@ def fit_transform(self, X, y):
       e = np.histogram_bin_edges(x, bins=self.n_bins)[1:]
       self.edges.append(e)
     return self._transform_after_preprocessor(X), y
-  def transform(self, X):
+  def transform(self, X, average=False):
     if self.preprocessor is not None:
       X = self.preprocessor.transform(X)
-    return self._transform_after_preprocessor(X)
-  def _transform_after_preprocessor(self, X):
-    fX = []
-    for j in range(X.shape[1]): # feature index
-      e = self.edges[j]
-      i_row = np.arange(X.shape[0])
-      i_col = np.clip(np.ceil((X[:,j] - e[0]) / (e[1]-e[0])).astype(int), 0, self.n_bins-1)
-      fX_j = csr_matrix(
-        (np.ones(X.shape[0], dtype=int), (i_row, i_col)),
-        shape = (X.shape[0], self.n_bins),
-      )
-      fX.append(fX_j.toarray())
-    fX = np.stack(fX).swapaxes(0, 1).reshape((X.shape[0], -1))
-    if self.unit_scale:
-      fX = fX / fX.sum(axis=1, keepdims=True)
-    return fX
+    return self._transform_after_preprocessor(X, average=average)
+  def _transform_after_preprocessor(self, X, average=False):
+    if not average:
+        return super()._transform_after_preprocessor(X) # FIXME
```

Comment:
At this point, `super()._transform_after_preprocessor` does not exist (it only existed in the performance test script where you implemented a subclass of the HistogramTransformer; here, the change affects the HistogramTransformer directly, so that `super()` is a different object).

Instead of  `super()._transform_after_preprocessor`, you need to put the original code of `_transform_after_preprocessor`, starting at the line with `fX = []` until `return fX`.

Reply:
Once you have made this change locally, please check that the unit tests of this package succeed. How to start the unit tests is described [in the documentation](https://mirkobunse.github.io/qunfold/developer-guide.html#unit-testing).

Reply:
Thanks for your feedback. I have edited the code as you suggested and run the unit test. I have also committed the edited code. Please let me know your opinion.